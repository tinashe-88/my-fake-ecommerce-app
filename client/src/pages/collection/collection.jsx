import React from 'react'
import { connect } from 'react-redux'

import CollectionItem from '../../components/collection-item/collection-item'

import { selectCollection } from '../../redux/shop/shop.selector'

import {
  CollectionPageContainer,
  TitleText,
  ItemsContainer
} from './collection.styles'

const CollectionPage = ({ collection }) => {
  const { title, items } = collection

  return (
    <CollectionPageContainer>
    <TitleText>{title}</TitleText>
      <ItemsContainer>
        {
          items.map(item =>
            <CollectionItem key={item.id} item={item} />
          )
        }
      </ItemsContainer>
    </CollectionPageContainer>
  )
}

const mapStateToProps = (state, ownProps) => ({
    collection: selectCollection(ownProps.match.params.collectionId)(state)
})

export default connect(mapStateToProps)(CollectionPage)