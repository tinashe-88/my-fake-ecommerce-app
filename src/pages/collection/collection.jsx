import React from 'react'
import { connect } from 'react-redux'

import { selectCollection } from '../../redux/shop/shop.selector'

import CollectionItem from '../../components/collection-item/collection-item'

import {
  CollectionPageContainer,
  TitleText,
  ItemsContainer
} from './collection.styles'

const CollectionPage = ({ collection }) => {
  const { items, title } = collection
  return (
    <CollectionPageContainer>
    <TitleText>{title}</TitleText>
      <ItemsContainer>
        {items.map(item => (
          <CollectionItem key={item.id} item={item} />
        ))}
      </ItemsContainer>
    </CollectionPageContainer>
  )
}

// Special selector that needs a part of state depending on URL param
const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state)
})

export default connect(mapStateToProps)(CollectionPage)
