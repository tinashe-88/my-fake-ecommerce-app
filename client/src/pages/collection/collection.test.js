import React from 'react'
import { shallow } from 'enzyme'

import { CollectionPage } from './collection'
import CollectionItem from '../../components/collection-item/collection-item'

describe('', () => {
  let wrapper
  let mockItems = [
    { id: 1 },
    { id: 2 },
    { id: 3 },
  ]

  beforeEach(() => {
    const mockCollection = {
      items: mockItems,
      title: 'Collection Test'
    }

    wrapper = shallow(<CollectionPage collection={mockCollection}/>)
  })

  it('Should render the CollectionPage component', () => {
    expect(wrapper.find('CollectionPage')).toMatchSnapshot()
  })

  it('Should render the same amount of CollectionItem as the collection array', () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length)
  })
})