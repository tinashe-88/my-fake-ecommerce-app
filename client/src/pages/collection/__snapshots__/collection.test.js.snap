// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the CollectionPage component 1`] = `null`;
